### ISSUE TITLE
Integration time and response time

### ISSUE THREAD

**Issue by Turtle49** [adafruit/Adafruit_CircuitPython_TCS34725#33]
When acquiring data sometimes the response time is twice as long as expected. For integration times >90ms it takes ~95ms for the TCS34725 to respond, and expected. For <90ms it takes twice as long (say, 186ms for integration time of 80ms).

This affects situations in a continuous loop, since for integration times < 90ms will have ~50% "sensing" rate. I've wrestled with the problem for a while. It seems to be less of an issue in M4 and in the latest TCS34725 driver (v3.3.3). In M0 and older drivers the issue persisted through all of integration times.

I used the code below for testing

import time
import board
import adafruit_tcs34725
sensorRGB = adafruit_tcs34725.TCS34725(board.I2C())
sensorRGB.integration_time =  90#for integration time 2.4 - 614.4 ms
while True:
    tic = time.monotonic()
    print(sensorRGB.color_raw)
    print(time.monotonic()-tic)

**Comment by siddacious** [comment]
Thanks for bringing this to our attention.

I took a quick glance and it looks like the library is waiting for the sensor's status register to report that it's "valid" meaning it's completed an integration cycle.

@Turtle49 it may be possible to return the "old" value if a new one isn't ready yet. Would that work for you? It may not be possible depending on how the sensor works and updates the measurement registers but it's worth looking into.

@dherrada do you want to take a look at this?

**Comment by Turtle49** [comment]
Yes, that may be the right solution @siddacious. I imagine the data was collected by the sensor but the microcroprocessor didn't "hear" (or wasn't "told") that the data is ready to be read and so the sensor gets it again (?), in which case the previous data will still be new to the microprocessor, because it never got it.

It is very peculiar that the wait time is roughly 2x the integration time. It suggests that the data is acquired twice, not just simply time-out, etc. But perhaps I am wrong.

PS: big thanks to this community for their contributions

**Comment by Turtle49** [comment]
I found a way to address the problem described above using in line 227 of adafruit_tcs34725.py. This line adds a pause to the loop that waits for the "valid" pin to become True. Currently the wait is about as long as the integration time set by the user. I changed it to a value of 1 ms. It works quite well in practice and it seems like an adequate permanent.

Again, I changed line 227 from: "time.sleep((self._integration_time + 0.9) / 1000.0)" to: "time.sleep(0.001)".
